### ISSUE TITLE
Multipart HTTP update support

### ISSUE THREAD

**Issue by ofaurax** [bmc-toolbox/bmclib#341]
When trying to update a firmware with bmclib (using install-firmware example):
```
$ ./main --host REDACTED --user root --password REDACTED --component bios --firmware update.tar.gz
FATA[0007] 3 errors occurred:
	* provider: gofish: Multipart HTTP push updates not supported: redfish update service error
	* Multipart HTTP push updates not supported: redfish update service error
	* failure in FirmwareInstall
```

**Comment by ofaurax** [comment]
It seems only multipart update is supported. OpenBMC only supports basic update, as of now.

**Comment by ofaurax** [comment]
> OpenBMC only supports basic update, as of now.

In fact, OpenBMC implements "Unstructured HTTP push update", which is deprecated, not "basic update".

**Comment by ofaurax** [comment]
We implemented "Unstructured HTTP push update" in bmclib so that OpenBMC can be updated

**Comment by ofaurax** [comment]
Also, support for multipart update has been merged in openbmc:
https://gerrit.openbmc.org/c/openbmc/bmcweb/+/32174

**Comment by ofaurax** [comment]
> Awesome! That means this should now work with bmclib if using the latest openbmc since bmclib already supports this, correct?

Yes, and if both are supported, bmclib should use multipart update. Please give feedback of you test it

**Comment by davidallendj** [comment]
I forgot to come back and report, but I have had success updating firmware after the multipart update support was added to OpenBMC.
